v1/nix/modules/drv-parts/lock/default.nix

```diff
@@ -85,39 +79,51 @@
                   with open(out_path, 'w') as f:
                       json.dump(checksum, f, indent=2)
                   exit(0)
-      print("Could not determine hash", file=sys.stdout)
-      exit(1)
+          process.wait()
+          if process.returncode:
+              print("Could not determine hash", file=sys.stdout)
+              exit(1)
+      # At this point the derivation was built successfully and we can just read
+      #   the hash from the drv file.
+      show_derivation = ["${config.deps.nix}/bin/nix", "show-derivation", drv_path]  # noqa: E501
+      result = subprocess.run(show_derivation, stdout=subprocess.PIPE)
+      drv = json.loads(result.stdout.decode())
+      checksum = drv[drv_path]["outputs"]["out"]["hash"]
+      checksum = codecs.encode(codecs.decode(checksum, 'hex'), 'base64').decode()
+      checksum = f"sha256-{checksum}".strip()
```

Comment:
Why do we need strip() here? Is the newline coming from the drv in line 91?

Reply:
codecs.encode adds this newline. I moved the strip to the upper line to make this more clear.